core/src/main/java/org/bitcoinj/wallet/DeterministicKeyChain.java

```diff
@@ -1174,20 +1176,27 @@ public int getLookaheadThreshold() {
     public void maybeLookAhead() {
         lock.lock();
         try {
-            List<DeterministicKey> keys = maybeLookAhead(externalParentKey, issuedExternalKeys);
-            keys.addAll(maybeLookAhead(internalParentKey, issuedInternalKeys));
-            if (keys.isEmpty())
-                return;
-            keyLookaheadEpoch++;
-            // Batch add all keys at once so there's only one event listener invocation, as this will be listened to
-            // by the wallet and used to rebuild/broadcast the Bloom filter. That's expensive so we don't want to do
-            // it more often than necessary.
-            basicKeyChain.importKeys(keys);
+            List<DeterministicKey> keys = concatLists(
+                    maybeLookAhead(externalParentKey, issuedExternalKeys),
+                    maybeLookAhead(internalParentKey, issuedInternalKeys));
+            if (!keys.isEmpty()) {
+                keyLookaheadEpoch++;
+                // Batch add all keys at once so there's only one event listener invocation, as this will be listened to
+                // by the wallet and used to rebuild/broadcast the Bloom filter. That's expensive so we don't want to do
+                // it more often than necessary.
+                basicKeyChain.importKeys(keys);
```

Comment:
Are these two lines not also a candidate for wrapping in a method?

Reply:
That's the next PR!

Reply:
That's the next PR!

Reply:
:smile: